backend/templates/__main__.mustache

```diff
@@ -16,23 +17,24 @@ class CorsHeaderMiddleware(object):
 
         return self.app(environ, custom_start_response)
 
-def main():
+
+def create_app():
     app = connexion.App(__name__, specification_dir='./swagger/')
     app.app.json_encoder = encoder.JSONEncoder
     app.add_api('swagger.yaml', arguments={'title': '{{appName}}'},
                     pythonic_params=True, validate_responses=True)
     app.app.wsgi_app = CorsHeaderMiddleware(app.app.wsgi_app)
-   
+
     FLASK_ENV = os.getenv('FLASK_ENV')
     if FLASK_ENV == 'production':
         app.app.config.from_object(ProductionConfig())
     elif FLASK_ENV == 'development':
         app.app.config.from_object(LocalDevelopmentConfig())
     elif FLASK_ENV == 'staging':
         app.app.config.from_object(StagingConfig())
-    
-    app.run(port={{serverPort}})
+
+    return app
 
 
 if __name__ == '__main__':
-    main()
+    create_app().run(port={{serverPort}})
```

Comment:
general comments, not a fix for this pull request:

It looks template variables are used to insert just variables (except `packageName`). I think these treatment is usually done by relying on environment variables. Are there any reasons to choose this way?

Reply:
Background: We use **swagger-codegen-cli** to generate template code for the backend and the frontend. All the templates are copy-pasted from their codebase. I had to make a few modifications (bug fixes and better type hints,  etc) and that's why these files are in this codebase. I don't want to change the templates too much in case we want to upgrade to a newer version of the templates in the future.

`packageName` and `serverPort` are both arguments of **swagger-codegen-cli** so they are template variables rather than environment variables.

Reply:
Thanks for clarification!

In several cases, we couldn't select the port number by ourselves because the runtime environment constrains it.
E.g., Cloud Run expects that the inner process listens on the specified port, whose number is provided as `PORT`.
Maybe strictly relying on official templates provided by the tools narrows our choice, especially deploying the service on IaaS which allows us to bring many automation at scale.

We may return to this topic in case we need some decisions related to runtime environment, but for now we can put this as-is.

Reply:
That's a good point! We currently hardcode 5000 as the port number for the backend in 4 places (Dockerfile, nginx.conf, config file for swagger-codegen-cli, and frontend proxy). If necessary, we can update them to read from an environment variable.